### ISSUE TITLE
How to make a target depends on arbitrary file being changed?

### ISSUE THREAD

**Issue by abailly** [ndmitchell/shake#204]
Hello,

I have the following rule in my shake  `Build.hs` file:

```
    "images/app.uuid" *> \uuidFile -> do
        cljs <- getDirectoryFiles "ui" ["//*.cljs"]
        hs <- getDirectoryFiles "src" ["//*.hs"]
        mapM_ readFile' (L.map ("ui" </>) cljs)
        mapM_ readFile' (L.map ("src" </>) hs)
        need ["images/deps.uuid", "Dockerfile", "ui/project.clj"]
        buildAppContainer uuidFile "capital" 
```

the `buildAppContainer` function builds a docker container and records its id for tracking. The idea is that we want to build the container if any of the source files has changed. The above fragment rebuilds the app all the time, even if no file has changed... 

I cannot find a way to say `fileChanged f`` to express dependency on some source file having changed. If  I add a `need [ hs ]` then I need a rule to "build" source files which does not make sense either.

Help greatly appreciated...

**Comment by ndmitchell** [comment]
Assuming the suggestions worked - if not please let me know.

**Comment by abailly** [comment]
Thanks and apologies for not having closed the issue :-(
Looks like things are working fine now

Le jeudi 12 février 2015, Neil Mitchell <notifications@github.com> a écrit :

> Closed #204 <https://github.com/ndmitchell/shake/issues/204>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ndmitchell/shake/issues/204#event-234767358>.
>


-- 
-- 
Arnaud Bailly

twitter: abailly
skype: arnaud-bailly
linkedin: http://fr.linkedin.com/in/arnaudbailly/
